cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/blobupload/StorageClient.java

```diff
@@ -216,7 +215,8 @@ private List<CompletedPart> uploadPartsOfBundle(String key, String uploadId,
             {
                 int loopPartNumber = partNumber;
                 ByteBuffer buffer = chunks.next();
-                AsyncRequestBody body = AsyncRequestBody.fromBytes(ByteBufferUtils.getArray(buffer));
+                // get a copy of the remaining ByteBuffer; for the last chunk, the cloned ByteBuffer is resized accordingly.
+                AsyncRequestBody body = AsyncRequestBody.fromRemainingByteBuffer(buffer);
```

Comment:
`AsyncRequestBody.fromBytes(ByteBufferUtils.getArray(buffer))` copies data 2 times.. 
1. copy from buffer to byte array
2. copy from byte array to another byte array in `AsyncRequestBody.fromBytes`

By switching to `AsyncRequestBody.fromRemainingByteBuffer`, it copies once, i.e. reduce 1 data copy